addons/mail/models/template_reset_mixin.py

```diff
@@ -0,0 +1,142 @@
+            fields_to_empty = fields_to_empty - set(field_defaults.keys())
+            # Finally, update the values with fields that should be empty
+            values.update({field: False for field in fields_to_empty})
+        return super()._load_records_write(values)
+
+    # -------------------------------------------------------------------------
+    # RESET TEMPLATE
+    # -------------------------------------------------------------------------
+
+    def _override_translation_term(self, module_name, xml_ids):
+        processed_base_langs = []
+        trans_to_remove = []
+        for code, _ in self.env['res.lang'].get_installed():
+            lang_code = tools.get_iso_codes(code)
+            is_po_file_available = False
+            # In case of sub languages (e.g fr_BE), load the base language first, (e.g fr.po) and
+            # then load the main translation file (e.g fr_BE.po)
+
+            # Step 1: reset translation terms with base language file
+            if '_' in lang_code:
+                base_lang_code = lang_code.split('_')[0]
+                base_trans_file = get_module_resource(module_name, 'i18n', base_lang_code + '.po')
+                if base_trans_file:
+                    is_po_file_available = True
+                    if base_lang_code not in processed_base_langs:
+                        processed_base_langs.append(base_lang_code)
+                        self._process_translation_data(base_trans_file, code, xml_ids)
+
+            # Step 2: reset translation file with main language file (can possibly override the
+            # terms coming from the base language)
+            trans_file = get_module_resource(module_name, 'i18n', lang_code + '.po')
+            if trans_file:
+                is_po_file_available = True
+                self._process_translation_data(trans_file, code, xml_ids)
+
+            # If no translation file available, unlink custom translated terms linked to template
+            if not is_po_file_available:
+                trans_to_remove.append(code)
+
+        if trans_to_remove:
+            self.env['ir.translation'].search([
+                ('name', 'like', f'{self._name},%'),
+                ('res_id', '=', self.id),
+                ('lang', 'in', trans_to_remove),
+                ('module', 'in', [module_name, False]),
+            ]).unlink()
+
+    def _process_translation_data(self, trans_file, lang, xml_ids):
+        """Populates the ir_translation table.
+        :param trans_file: path to a translation file to open, e.g. {addon_path}/mail/i18n/es.po
+        :param lang: language code of the translations contained in `trans_file`
+                     language must be present and activated in the database
+        """
+        with file_open(trans_file, mode='rb', filter_ext=(".po",)) as fileobj:
+            fileformat = os.path.splitext(trans_file)[-1][1:].lower()
+            reader = TranslationFileReader(fileobj, fileformat=fileformat)
```

Comment:
```suggestion
            reader = TranslationFileReader(fileobj)
```
`po` is the default and the `file_open` is getting filtered on the `po`.